File: src/main/java/kr/hahaha98757/zombiesaddon/data/Window.java
// Code in Zombies Utils by Stachelbeere1248

package kr.hahaha98757.zombiesaddon.data;

public class Window {
	private final short[] xyz = new short[3];
	private final byte id;
	private boolean isActive;

	public Window(int id, int x, int y, int z) {
		this.id = (byte) id;
		xyz[0] = (short) x;
		xyz[1] = (short) y;
		xyz[2] = (short) z;
	}

	public int getID() {
		return id;
	}

	public short[] getXYZ() {
		return xyz;
	}

	public void setActive(boolean active) {
		isActive = active;
	}

	public boolean isActive() {
		return isActive;
	}
}

File: src/main/kotlin/kr/hahaha98757/zombiesaddon/data/Window.kt
package kr.hahaha98757.zombiesaddon.data

class Window(val id: Int, private var x: Short, private var y: Short, private var z: Short) {
    var active = false
    val xyz get() = intArrayOf(x.toInt(), y.toInt(), z.toInt())

    fun rotate(rotations: Int) {
        when (rotations) {
            1 -> rotateCounterClockwise()
            2 -> mirrorBoth()
            3 -> rotateClockwise()
        }
    }

    fun mirrorX() {
        x = (-x).toShort()
    }

    fun mirrorZ() {
        z = (-z).toShort()
    }

    private fun rotateCounterClockwise() {
        val temp = x
        x = (-z).toShort()
        z = temp
    }

    private fun mirrorBoth() {
        x = (-x).toShort()
        z = (-z).toShort()
    }

    private fun rotateClockwise() {
        val temp = x
        x = z
        z = (-temp).toShort()
    }
}